modules/plottinggl/src/processors/colorscalelegend.cpp

```diff
@@ -220,11 +221,9 @@ std::tuple<ivec2, ivec2, ivec2, ivec2> ColorScaleLegend::getPositions(ivec2 dime
 }
 
 void ColorScaleLegend::process() {
-    // draw cached overlay on top of the input image
-    if (inport_.isReady()) {
-        utilgl::activateTargetAndCopySource(outport_, inport_);
-    } else {
-        utilgl::activateAndClearTarget(outport_);
+    utilgl::activateTargetAndClearOrCopySource(outport_, inport_);
```

Comment:
No need to copy data?
outport_.setData(inport_.getData());

Reply:
nope due to activating and copying/clearing the canvas